src/main/java/io/vertx/spi/cluster/hazelcast/impl/HazelcastAsyncMultiMap.java

```diff
@@ -96,35 +93,74 @@ public void add(K k, V v, Handler<AsyncResult<Void>> completionHandler) {
 
   @Override
   public void get(K k, Handler<AsyncResult<ChoosableIterable<V>>> resultHandler) {
-    ChoosableSet<V> entries = cache.get(k);
-    if (entries != null && entries.isInitialised() && getInProgressCount.get() == 0) {
-      resultHandler.handle(Future.succeededFuture(entries));
-    } else {
-      getInProgressCount.incrementAndGet();
-      vertx.getOrCreateContext().<ChoosableIterable<V>>executeBlocking(fut -> {
-        Collection<V> entries2 = map.get(k);
-        ChoosableSet<V> sids;
-        if (entries2 != null) {
-          sids = new ChoosableSet<>(entries2.size());
-          for (V hid : entries2) {
-            sids.add(hid);
-          }
-        } else {
-          sids = new ChoosableSet<>(0);
+    ContextImpl context = vertx.getOrCreateContext();
```

Comment:
shouldn't we use instead `Vertx.currentContext`, when there is no context the `EventBus` shall use the `sendNoContext` so it should be good.